eclipse-collections/src/main/java/org/eclipse/collections/impl/map/strategy/immutable/ImmutableUnifiedMapWithHashingStrategy.java

```diff
@@ -137,9 +136,9 @@ public void forEachKeyValue(Procedure2<? super K, ? super V> procedure)
     }
 
     @Override
-    public Set<K> keySet()
+    public ImmutableKeySet<K> keySet()
     {
-        return UnmodifiableMutableSet.of(this.delegate.keySet());
+        return new ImmutableKeySet<K>(this.delegate.keySet().toImmutable()){};
```

Comment:
@johnjstuart I wonder if we could wrap the mutable keySet here with an Immutable wrapper.

Reply:
@johnjstuart isn't `ImmutableKeySet` already behaving immutably? Why call `toImmutable` on top of this?